.github/workflows/build.yml

```diff
@@ -63,16 +64,41 @@ jobs:
         timeout_minutes: 10
         max_attempts: 3
         retry_wait_seconds: 10
-        command: docker pull docker.pkg.github.com/apache/incubator-nuttx-testing/nuttx-ci-linux
+        command: docker pull $IMAGE_TAG
 
+    - name: Restore ccache
+      id: ccache
+      uses: actions/cache@v2
+      with:
+        path: ccache
+        key: ccache-${{ runner.os }}-${{matrix.boards}}-
+        restore-keys: ccache-${{ runner.os }}-${{matrix.boards}}-${{ github.run_id }}
     - name: Run builds
       uses: ./testing/.github/actions/ci-container
       env:
         BLOBDIR: /tools/blobs
       with:
         run: |
+          apt-get update -qq && apt-get install -y -qq --no-install-recommends ccache
```

Comment:
Note the installation of ccache and the symbolic links are only here because the run will not make use of the changes in the Docker image.  Once this is ready they will be dropped from here and only exist in the image.